distributed/client.py

```diff
@@ -3960,6 +3960,31 @@ async def _get_task_stream(
         else:
             return msgs
 
+    def register_scheduler_plugin(self, plugin=None, idempotent=False, **kwargs):
+        """Register a scheduler plugin.
+
+        See https://distributed.readthedocs.io/en/latest/plugins.html
+
+        Parameters
+        ----------
+        plugin : SchedulerPlugin
+            Plugin object to pass to the scheduler.
+        idempotent : bool
+            If multiple registrations of the plugin induce no changes.
+
+        """
+
+        async def f(dask_scheduler=None, plugin=None, idempotent=None, **kwargs):
+            if isinstance(plugin, type):
+                plugin = plugin(**kwargs)
+            await plugin.start()
+            dask_scheduler.add_plugin(plugin=plugin, idempotent=idempotent, **kwargs)
+
+        self.run_on_scheduler(f, plugin=plugin, idempotent=idempotent, **kwargs)
```

Comment:
I would suggest to define a new _handler_ in the scheduler (see `Scheduler.handlers`, there is also the same thing for the worker plugin) and the logic of `f` is then executed by the handler. This way, the client call here is actually quite trivial

Defining `f` in this local scope here is indeed sometimes problematic since it sometimes introduces serializability issues and a global scope (class method or global function) is usually better when serializing stuff